cou/steps/plan.py

```diff
@@ -55,14 +58,51 @@
 logger = logging.getLogger(__name__)
 
 
+class PlanWarnings:
+    """Representation of a collection of warning messages from plan generation.
+
+    This is a singleton class which holds all warning messages returned by applications
+    when generating a plan.
+    """
+
+    _instance = None
+
+    def __new__(cls) -> PlanWarnings:
+        """Create a new instance of the PlanWarnings class if doesn't already exist."""
+        if not cls._instance:
+            cls._instance = super().__new__(cls)
+        return cls._instance
+
+    def __init__(self) -> None:
+        """Initialize the PlanWarnings class."""
+        self._warnings: list[str] = []
+
+    def add_warning(self, warning: str) -> None:
+        """Add a new warning to the collection.
+
+        :param warning: A warning message to be stored.
+        :type warning: str
+        """
+        self._warnings.append(warning)
+
+    @property
+    def warnings(self) -> list[str]:
+        """Get all warnings stored in the class.
+
+        :return: The list of all warnings stored in the class.
+        :rtype: list[str]
+        """
+        return self._warnings
```

Comment:
I have some concerns with this class that I would to check:

I did some test and it's not behaving as I expect:

```terminal
In [117]: a = PlanWarnings()

In [118]: b = PlanWarnings()


In [119]: a is b
Out[119]: True

In [120]: a.add_warning("asdf")

In [122]: a.warnings
Out[122]: ['asdf']

In [123]: b.warnings
Out[123]: ['asdf']

In [125]: PlanWarnings().add_warning("1")

In [126]: b.warnings
Out[126]: ['1']

In [127]: PlanWarnings().add_warning("3")

In [128]: b.warnings
Out[128]: ['3']

In [129]: PlanWarnings().warnings
Out[129]: []
```
It seems that every time that PlanWarnings().warnings is called a new `_warnings` is created, so if we use this on other modules or in different executions of `_generate_instance_plan` we won't have a single warning messages

Reply:
Good catch, I was thinking about it seeing `__init__` implementation, because of `__init__` defining empty warnings, those should be defined in class instead of _instance?

Reply:
`_instance` is also needed to to hold that single instance whenever the user tries to instantiate.

Reply:
Resolving this thread now that we simply use class method